File: bounce/src/main/java/is/ricko/bounce/config/BounceAppListener.java
package is.ricko.bounce.config;

import lombok.extern.java.Log;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.stereotype.Component;

@Component
@Log
public class BounceAppListener implements ApplicationListener<ApplicationReadyEvent> {
  @Override
  public void onApplicationEvent(final ApplicationReadyEvent event) {
    log.info("Boot complete. Signalling GC.");
    System.gc();
    final String port = event.getApplicationContext().getEnvironment().getProperty("local.server.port", "");
    log.info("App available at:\nhttp://localhost:" + port);
  }
}

File: bounce/src/main/kotlin/is/ricko/bounce/config/BounceAppListener.kt
package `is`.ricko.bounce.config

import org.slf4j.Logger
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.ApplicationListener
import org.springframework.stereotype.Component

@Component
class BounceAppListener : ApplicationListener<ApplicationReadyEvent> {
    @Logging
    private var logger: Logger? = null

    override fun onApplicationEvent(event: ApplicationReadyEvent) {
        logger?.info("Boot complete. Signalling GC.")
        System.gc()
    }
}